### PR TITLE
Start importing all footways and shared-use paths! #87, #77

### PR DESCRIPTION
**Pull request by dabreegster**
This introduces two new lane types and starts importing from OSM. The experiment for separate sidewalk handling is refined to only deal with parallel sidewalks and crossings. All the footways are imported always.

I'm immensely excited by this PR. Previously, most footways were either not imported at all or mangled into bike lanes. This is a huge step forwards that doesn't get stuck on the problem of parallel sidewalks. Some demos:

Screenshot from 2022-09-15 10-59-05
Screenshot from 2022-09-15 10-58-38
Screenshot from 2022-09-15 10-58-08

I've validated with the test explorer, clicking through to open the OSM ways and spot-check things. I'm next going to update A/B Street with these changes and work through integration issues there.

### CODE REVIEW

File: import_streets/src/extract.rs [hunk 1]
```diff
@@ -119,15 +119,14 @@ impl OsmExtract {
             return false;
         }
 
-        // If we're only handling sidewalks tagged on roads, skip a bunch of ways
+        // If we're only handling sidewalks tagged on roads, skip crossings and separate sidewalks
+        // Note we have to do this here -- get_lane_specs_ltr doesn't support decisions like
+        // "actually, let's pretend this road doesn't exist at all"
         if opts.map_config.inferred_sidewalks {
```
Comment (dabreegster) [inline import_streets/src/extract.rs:125]: So this experiment now just means "bring in `highway=footway, footway={sidewalk,crossing}` ways". All of the new stuff in this PR works both with and without this experiment

File: street_network/src/lanes/classic.rs [hunk 2]
```diff
@@ -31,33 +34,20 @@ pub fn get_lane_specs_ltr(tags: &Tags, cfg: &MapConfig) -> Vec<LaneSpec> {
...
-        // allowed.
-        if tags.is("bicycle", "no")
-            || (tags.is(osm::HIGHWAY, "footway")
-                && !tags.is_any("bicycle", vec!["designated", "yes", "dismount"]))
-        {
-            return vec![fwd(LaneType::Sidewalk)];
-        }
-        // Otherwise, there'll always be a bike lane.
 
+    // If it's a primarily cycleway, have directional bike lanes and add a shoulder for walking
```
Comment (dabreegster) [inline street_network/src/lanes/classic.rs:38]: It's still odd to use two directional lanes for this case. I think `street_network::Direction` and everything downstream needs to have a both-ways case. It's a very complex change downstream, affecting routing, rendering, traffic simulation... so not attempting it yet.

File: street_network/src/lanes/mod.rs [hunk 3]
```diff
@@ -32,6 +32,11 @@ pub enum LaneType {
     Construction,
     LightRail,
     Buffer(BufferType),
+    /// Some kind of pedestrian-only path unassociated with a road
```
Comment (dabreegster) [inline street_network/src/lanes/mod.rs:35]: Very related to #77. I'm not happy with this classification as "only pedestrians" and "both pedestrians and cyclists." Things like living streets or alleyways, where motor vehicles are also allowed, aren't captured yet. Maybe we just need one shared use lane type, with a way to specify the legal or cultural priority between vehicles. For instance, I know many streets in the US without pavement, where it's effectively shared use and motor vehicles get priority.

File: street_network/src/lanes/mod.rs [hunk 4]
```diff
@@ -76,17 +83,22 @@ impl LaneType {
...
+        matches!(
+            self,
+            LaneType::Sidewalk | LaneType::Shoulder | LaneType::Footway | LaneType::SharedUse
+        )
     }
 
     pub fn describe(self) -> &'static str {
         match self {
             LaneType::Driving => "a general-purpose driving lane",
-            LaneType::Biking => "a protected bike lane",
```
Comment (dabreegster) [inline street_network/src/lanes/mod.rs:89]: "protected" never ever made sense, I erroneously wrote this long ago.

Also this is a localization issue... "cycling lane" or "cycle lane" would fit better in the UK. But anyway.

File: street_network/src/lanes/mod.rs [hunk 5]
```diff
@@ -154,6 +172,8 @@ impl LaneType {
             LaneType::Construction => 'x',
             LaneType::LightRail => 'l',
             LaneType::Buffer(_) => '|',
+            LaneType::Footway => 'f',
+            LaneType::SharedUse => 'V',
```
Comment (dabreegster) [inline street_network/src/lanes/mod.rs:176]: There was not an obvious choice here :(

File: tests-web/www/js/layers.js [hunk 6]
```diff
@@ -66,6 +66,25 @@ export const makeLanePolygonLayer = (text) => {
 
   return new L.geoJSON(JSON.parse(text), {
     style: function (feature) {
+      if (feature.properties.type == "Footway") {
```
Comment (dabreegster) [inline tests-web/www/js/layers.js:69]: The styling is not great. Any ideas for distinguishing the two?

File: street_network/src/lanes/mod.rs (hunk 3 above)
Comment (BudgieInWA) [reply street_network/src/lanes/mod.rs:35]: I think when it comes down to it, every lane is a list of specific designations and restrictions. There are also classifications, informed by the locale, I suppose, like "only pedestrians", "bike lane" or "bus lane (bikes allowed)" that inform markup decisions.

File: street_network/src/lanes/mod.rs (hunk 5 above)
Comment (BudgieInWA) [reply street_network/src/lanes/mod.rs:176]: Large '`F`'? It is appropriate that is suggests footway, I think.

File: street_network/src/lanes/classic.rs (hunk 2 above)
Comment (BudgieInWA) [reply street_network/src/lanes/classic.rs:38]: Yes, I agree. An "alternates" case too.

Most reads of the value will boil down to "is travel supported in this direction", right? Routing could implement "alternating" and use that for "both ways" too.

I called it [`TravelDirection` in my types](https://github.com/a-b-street/osm2streets/blob/a19ab6ac8ef1dc4b812830fc1eae11687e28356c/osm2streets/src/units.rs#L22-L33).

File: tests-web/www/js/layers.js (hunk 6 above)
Comment (BudgieInWA) [reply tests-web/www/js/layers.js:69]: Maybe it's time to start copying design from the [strassenraumkarte-neukoelln](https://github.com/SupaplexOSM/strassenraumkarte-neukoelln) :P

File: street_network/src/lanes/mod.rs (hunk 3 above)
Comment (dabreegster) [reply street_network/src/lanes/mod.rs:35]: I like this way of thinking about it. Going to start a fresh issue on this soon to consolidate all of the thoughts around this...

File: tests-web/www/js/layers.js (hunk 6 above)
Comment (dabreegster) [reply tests-web/www/js/layers.js:69]: I already lifted the dashed outline idea and path color from there ;)

**Comment by dabreegster** [comment]
https://github.com/a-b-street/abstreet/pull/994 has more rendering demo. Mostly smooth sailing there, I expect to finish ironing out some integration problems and merge both in a few hours!

**Pull request merged**